autoload/phpactor.vim

```diff
@@ -435,14 +435,41 @@ endfunction
 " RPC -->-->-->-->-->--
 """""""""""""""""""""""
 
+let g:phpactorProjectRootPatterns = ['composer.json', '.git']
+let g:phpactorGlobalRootPatterns = ['/', '/home']
+
+function! s:searchDirectoryUpwardForRootPatterns(initialDirectory, rootPatterns, fallbackDirectory)
+  let l:directory = a:initialDirectory
+  while index(g:phpactorGlobalRootPatterns, l:directory) < 0
+    for l:pattern in a:rootPatterns
+      if (filereadable(l:directory .'/'. l:pattern))
+        return l:directory
+      endif
+    endfor
+    let l:directory = fnamemodify(l:directory, ':h')
+  endwhile
+
+  if index(g:phpactorGlobalRootPatterns, l:workspaceDir) >= 0
```

Comment:
`l:workspaceDir` is a local var? How is it defined at this point?

Reply:
`l:directory` should be there, naturally :+1: